### PR TITLE
Remove most of libs/rand

### PR DESCRIPTION
**Pull request by liamsi**
Cherrypicked the changes from https://github.com/tendermint/tendermint/pull/6364 into this repo.
Context: https://github.com/tendermint/tendermint/issues/2956

math/rand's global object is already go-routines safe: https://github.com/golang/go/blob/fd3612e4339d10a0e02571a8e43868d569b11a9f/src/math/rand/rand.go#L293
https://github.com/golang/go/blob/fd3612e4339d10a0e02571a8e43868d569b11a9f/src/math/rand/rand.go#L382

There is no need to maintain a copy of this in this repo.

Partly triggered by the discussion in https://github.com/lazyledger/lazyledger-core/pull/270#issuecomment-820339622

### CODE REVIEW

File: types/block_test.go [hunk 1]
```diff
@@ -698,7 +698,7 @@ func TestBlockProtoBuf(t *testing.T) {
 		if tc.expPass2 {
 			require.NoError(t, err, tc.msg)
 			require.EqualValues(t, tc.b1.Header, block.Header, tc.msg)
-			// require.EqualValues(t, tc.b1.Data, block.Data, tc.msg)
```
Comment (liamsi) [inline types/block_test.go:701]: Interesting. Why did we comment that one out?

File: libs/rand/random.go [hunk 2]
```diff
@@ -165,147 +61,29 @@ MAIN_LOOP:
...
-	return time.Unix(int64(r.Uint64()), 0)
-}
-
-// Bytes returns n random bytes generated from the internal
-// prng.
-func (r *Rand) Bytes(n int) []byte {
-	// cRandBytes isn't guaranteed to be fast so instead
-	// use random bytes generated from the internal PRNG
+// Bytes returns n random bytes generated from a freshly instantiated prng.
+func Bytes(n int) []byte {
```
Comment (Wondertan) [inline libs/rand/random.go:73]: Prng here does not seem to be freshly generated :)

Comment (liamsi) [reply libs/rand/random.go:73]: Good catch, will update the comment.

File: test/e2e/runner/perturb.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ func PerturbNode(node *e2e.Node, perturbation e2e.Perturbation) (*rpctypes.Resul
 		return nil, fmt.Errorf("unexpected perturbation %q", perturbation)
 	}
 
-	status, err := waitForNode(node, 0, 10*time.Second)
+	status, err := waitForNode(node, 0, 20*time.Second)
```
Comment (evan-forbes) [inline test/e2e/runner/perturb.go:69]: Out of curiosity, was this needed specifically for this PR or just in general?

**Review by evan-forbes — APPROVED** [review]

File: test/e2e/runner/perturb.go (hunk 3 above)
Comment (liamsi) [reply test/e2e/runner/perturb.go:69]: It's completely unrelated to this PR. I hope with this and with #295 e2e test will pass most of the times 🤞🏼

**Pull request merged**